ProcessLib/LocalAssemblerInterface.h

```diff
@@ -26,8 +26,20 @@ class LocalAssemblerInterface
 
     virtual void assemble(double const t, std::vector<double> const& local_x) = 0;
 
-    virtual void addToGlobal(AssemblerLib::LocalToGlobalIndexMap::RowColumnIndices const&,
-            GlobalMatrix& M, GlobalMatrix& K, GlobalVector& b) const = 0;
+    virtual void assembleJacobian(double const t,
+                                  std::vector<double> const& local_x) = 0;
+
+    virtual void preTimestep(std::vector<double> const& local_x) = 0;
+    virtual void postTimestep(std::vector<double> const& local_x,
+                              double const t) = 0;
```

Comment:
Yes I thought about it, but the default implementation might have a drawback that these functions are called for every element. I'll need to check that before providing the defaults here. Tonight.